test/src/test/java/org/corfudb/integration/StreamingIT.java

```diff
@@ -1396,6 +1396,77 @@ public void testSubscribeListenerNoTimestamp() throws Exception {
         assertThat(listenerPostWrites.getUpdates().size()).isEqualTo(numUpdates);
     }
 
+    /**
+     * Test that updates to the same key within a transaction boundary does not lead
+     * to multiple stream notifications, i.e., duplication.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testDeduplicationOfTxUpdates() throws Exception {
+        // Run a corfu server.
+        Process corfuServer = runSinglePersistentServer(corfuSingleNodeHost, corfuStringNodePort);
+
+        // Start a Corfu runtime.
+        runtime = createRuntime(singleNodeEndpoint);
+        runtime.setTransactionLogging(true);
+
+        CorfuStore store = new CorfuStore(runtime);
+
+        final String namespace = "test_namespace";
+        final String tableName = "table";
+        final int numUpdates = 20;
+        final int delta = 30;
+
+        // Create a table.
+        Table<Uuid, SampleTableAMsg, Uuid> tableA = store.openTable(
+                namespace, tableName,
+                Uuid.class, SampleTableAMsg.class, Uuid.class,
+                TableOptions.builder().build()
+        );
+
+        // Subscribe to streaming updates
+        StreamListenerImpl listener = new StreamListenerImpl("stream_listener");
+        store.subscribeListener(listener, namespace, "sample_streamer_1", Collections.singletonList(tableName));
+
+        // Update same key multiple times within the same transaction
+        try (TxnContext tx = store.txn(namespace)) {
+            Uuid key1 = Uuid.newBuilder().setMsb(0).setLsb(0).build();
+
+            // Update key 1 multiple times within the same transaction
+            for (int i = 0; i < numUpdates; i++) {
+                SampleTableAMsg msg = SampleTableAMsg.newBuilder().setPayload(String.valueOf(i)).build();
+                tx.putRecord(tableA, key1, msg, key1);
+            }
+
+            // Add another (unique) key to confirm order of updates is preserved
```

Comment:
not sure I understand this comment... no order is guaranteed within a transaction, right?

Reply:
we had a discussion and decided to change the behavior, so this was just left behind, changing it! thanks!